File: codes/yunyeong/560_subarraysumequlask.py
class Solution:
    def subarraySum(self, nums: list[int], k: int) -> int:
        arr=[]
        for i in range(len(nums)):
            temp =[]
            for j in range(i,len(nums)):
                if nums[j] <= k:
                    temp.append(nums[j])
                    if k == sum(temp):
                        arr.append(temp)
                        break
        return len(arr)

# - was not considered.

File: codes/naeun/560_subarraysumequlask.java
class Solution {

    public int subarraySum(int[] nums, int k) {
        // answer is total num of subarrays whose sum == k
        int answer = 0;

        int rightSum = 0;
        for(int right = 0; right< nums.length; right++){
            rightSum += nums[right]; 
            
            int totalSum = rightSum;
            for(int left = 0; left <= right; left ++){
                if(totalSum == k){
                    answer += 1;
                }
                totalSum -= nums[left];
            }
        }
        // return answer
        return answer;
    }

}